### ISSUE TITLE
CMake installation issue: Most users just want `vk_mem_alloc.h`

### ISSUE THREAD

**Issue by juan-lunarg** [GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator#346]
I think this issue is worth bringing up since I've seen the same pattern in 2 places. When shipping the Vulkan SDK and with vcpgk.

NOTE: I believe Conan also doesn't use the default installation either (https://github.com/conan-io/conan-center-index/tree/master/recipes/vulkan-memory-allocator/all). But I'm not as familiar with Conan.

### Problem: Most users just want `vk_mem_alloc.h`

Here is the current port code for vcpkg:
```
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator
    REF a6bfc237255a6bac1513f7c1ebde6d8aed6b5191 #v3.0.1
    SHA512  14361ff201fd660c22b60de54c648ff20a2e2a7f65105f66853a9a4dbffbeca2ae42098dcb1528bb4e524639b92fa4ff27ebd3940c42ccfaf7c99c08bdd0d8ce
    HEAD_REF master
    PATCHES
        gcc13.patch # https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/commit/29d492b60c84ca784ea0943efc7d2e6e0f3bdaac
)

file(COPY "${SOURCE_PATH}/include/vk_mem_alloc.h" DESTINATION "${CURRENT_PACKAGES_DIR}/include")

file(COPY "${CMAKE_CURRENT_LIST_DIR}/unofficial-vulkan-memory-allocator-config.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/unofficial-vulkan-memory-allocator")

configure_file("${SOURCE_PATH}/LICENSE.txt" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
```

It's ignoring the installation method natively provided by `VulkanMemoryAllocator` and providing it's own method instead.

Same applies to the Vulkan SDK. We also ignore the installation provided by `VulkanMemoryAllocator` and just manually copy the include file. We don't want to ship binaries for the `VulkanMemoryAllocator`.

**Comment by adam-sawicki-a** [comment]
VMA is designed as a single-header library and so I agree that it is recommended to just take the "vk_mem_alloc.h" file instead of building it as a separate library.

What is your request here? What do you propose to change in VMA repository?

**Comment by juan-lunarg** [comment]
> What is your request here? What do you propose to change in VMA repository?

Ideally VMA would behave like Vulkan-Headers and install only the header files. See the CMake code for Vulkan-Headers for reference.

https://github.com/KhronosGroup/Vulkan-Headers

**Comment by adam-sawicki-a** [comment]
VMA is intended to be used as a single self-contained file to be included in user's project. Cmake script was an external contribution. Since it was accepted, it became a never-ending story of issues and pull requests proposed to modify it. I would prefer to not refactor it so significantly. I will consider removing it completely in the future.

The script you showed above which does `vcpkg_from_github` can probably just `#include ".../vk_mem_alloc.h"` file wherever needed in the main project code without the need for the library to have its own Cmake script.

**Comment by juan-lunarg** [comment]
> VMA is intended to be used as a single self-contained file to be included in user's project.

I agree. Currently however it also installs binaries. This is a problem **regardless** of what build system VMA uses. This isn't a CMake exclusive problem this is a packaging problem.

```
juan@pop-os:~/projects/vma$ cmake --install build/ --prefix build/install
-- Installing: /home/.../install/lib/libVulkanMemoryAllocator.a <- This is the issue.
-- Installing: /home/.../install/include/vk_mem_alloc.h
```

> The script you showed above which does vcpkg_from_github can probably just #include ".../vk_mem_alloc.h" file wherever needed in the main project code without the need for the library to have its own Cmake script.

That is basically what vcpkg/conan and the vulkan-sdk do to avoid installing the aforementioned binary.

**Comment by juan-lunarg** [comment]
Closing since it seems like CMake support will be removed in the future.

**Comment by juan-lunarg** [comment]
Re-opening due to misinterpreting conversation.
